### ISSUE TITLE
ODHack: Fix and Complete Airdrop Tests

### ISSUE THREAD

**Issue by tensojka** [CarmineOptions/konoha#94]
## Description
The current `airdrop_tests.cairo` file contains quasi-broken tests, with many tests commented out. The goal is to make these tests functional. Reference the `defispring` repository, which contains a similar, tested contract developed based on the current airdrop contract.

### Tasks
1. **Uncomment and Fix Tests in `airdrop_tests.cairo`**

2. **Reference `defispring` Repository**:
   - Use the `defispring` repository as a reference to understand the testing approach.
   - Ensure the tests in `airdrop_tests.cairo` exceed the robustness and coverage of those in `defispring`.

3. **Generate Test Data Using `carmine-api-airdrop`**:
   - Utilize the `carmine-api-airdrop` to create the Merkle tree and generate proofs.
   - Use this test data in the `airdrop_tests.cairo`.

## Resources
- **Airdrop Tests (quasi-broken)**: [airdrop_tests.cairo](https://github.com/CarmineOptions/konoha/blob/master/tests/airdrop_tests.cairo)
- **Airdrop Component Code**: [airdrop.cairo](https://github.com/CarmineOptions/konoha/blob/master/src/airdrop.cairo)
- **Similar Tested Contract (defispring)**: [defispring Repository](https://github.com/starknetfndn/defispring)
- **Merkle Tree and Proof Generation**: [carmine-api-airdrop](https://github.com/CarmineOptions/carmine-api/tree/master/carmine-api-airdrop)
- **Test Cases Description**: [README](https://github.com/CarmineOptions/konoha/blob/master/tests/README.md)
- **Testing utils**: https://github.com/CarmineOptions/konoha/blob/master/tests/setup.cairo

## Acceptance Criteria
- All tests in `airdrop_tests.cairo` are uncommented, functional, and pass.
- Tests cover all specified cases in the [README](https://github.com/CarmineOptions/konoha/blob/master/tests/README.md).
- Test data generated using `carmine-api-airdrop` is correctly integrated into the tests.
- `scarb fmt`

Please make sure to keep the testing structure consistent and ensure the tests are comprehensive and robust. Use functions from `setup.cairo` and do not duplicate them, extend if needed.

**Comment by nicosanc** [comment]
Yes, I'd like to take on this assignment. Thank you for the opportunity!

**Comment by TropicalDog17** [comment]
Hi, could I work on this issue? 

I'm relatively new to Cairo, and I believe the best way to start learning a new codebase is from writing and fixing tests

**Comment by tensojka** [comment]
@TropicalDog17 Yes, good luck!

**Comment by TropicalDog17** [comment]
Yeah sorry I don't have the time to finish the task, feel free to assign someone else to do it, thanks!

**Comment by kateberryd** [comment]
@tensojka i would love to work on this. If it’s still open

**Comment by BenFaruna** [comment]
Hello @tensojka if this issue is still open I would like to work on it.

**Comment by tensojka** [comment]
Please fill out the application @BenFaruna .

@kateberryd I would prefer to have you available for other frontend tasks as you have experience with that.

**Comment by BenFaruna** [comment]
> Please fill out the application @BenFaruna .

I'm unable to apply from OnlyDust, the issue does not come up for projects except it's time for OD hack and the project is registered for that hack season.

**Comment by tensojka** [comment]
I see your aplication @BenFaruna but it doesn't show any previous OnlyDust experience. Our capacity to onboard new contributors is full at the moment, sorry.

**Comment by raizo07** [comment]
@tensojka can I work on this?

**Comment by BenFaruna** [comment]
@tensojka I contributed to a project earlier, but the project wasn't registered on Only dust so it doesn't show on the dashboard.

https://github.com/carbonable-labs/baseline-contracts/pull/3

That's my PR that got merged
